internal/aez/ccd_data.go

```diff
@@ -0,0 +1,170 @@
+package aez
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"math"
+	"time"
+
+	"github.com/innosat-mats/rac-extract-payload/internal/ccsds"
+)
+
+// WDWMode describes the CCD WDW parameter
+type WDWMode uint8
+
+// WDWModeManual = 0, check WDWOV for overflows
+//
+// WDWModeAutomatic, window selected depending on input data
+const (
+	WDWModeManual    WDWMode = 0
+	WDWModeAutomatic WDWMode = 1
+)
+
+// Wdw is the composite status of Image Window Mode
+type Wdw uint8
+
+// Mode returns the WDWMode type used encoded in Bit[7]
+func (wdw Wdw) Mode() WDWMode {
+	if (wdw & 1) == 0 {
+		return WDWModeManual
+	}
+	return WDWModeAutomatic
+}
+
+// InputDataWindow returns which bits out of the depth of the CCD Images
+//
+// It ueses information in Bit[2..0]
+//
+// Returns major and minor bit of original CCD Image and error
+// if encoding not covered by specification.
+//
+// If full range of Bit[15..0] is used the JPEGQ should be 0xFF
+func (wdw Wdw) InputDataWindow() (int, int, error) {
+	switch wdw >> 5 {
+	case 0x0:
+		return 11, 0, nil
+	case 0x1:
+		return 12, 1, nil
+	case 0x2:
+		return 13, 2, nil
+	case 0x3:
+		return 14, 3, nil
+	case 0x4:
+		return 15, 4, nil
+	case 0x7:
+		return 15, 0, nil
+	}
+	return -1, -1, fmt.Errorf(
+		"WDW value has unknown Input Data Window '%x'", wdw>>5,
+	)
+}
+
+// NCBin contains the FPGA and CCD columns bin count
+type NCBin uint16
+
+// FPGAColumns returns number FPGA columns to bin
+func (ncBin NCBin) FPGAColumns() int {
+	return int(math.Round(math.Pow(2, float64((ncBin<<4)>>8))))
+}
+
+// CCDColumns returns number of CCD columns to bin
+func (ncBin NCBin) CCDColumns() int {
+	return (int)(ncBin >> 8)
+}
+
+// CCDGain is game composite information
+type CCDGain uint16
+
+// CCDGainMode is high/low signal mode
+type CCDGainMode int
+
+// HighSignalMode, LowSignalMode
+const (
+	HighSignalMode CCDGainMode = iota
+	LowSignalMode
+)
+
+// CCDGainTiming is the timing flag
+type CCDGainTiming int
+
+// FasterTiming = 0, used for binned and discarded pixels
+//
+// FullTiming = 1, used even for pixels that are not read out
+const (
+	FasterTiming CCDGainTiming = iota
+	FullTiming
+)
+
+// Mode returns high/low signal mode
+func (gain CCDGain) Mode() CCDGainMode {
+	if gain&(1>>3) == 0 {

```

Comment:
(1>>3) :  shift a 1 three times to the right will always be zero. Maybe you want to change direction of the shift?
```
0b0000 00001 << 3 = 0b0000 0100
```